synapse/rest/saml2/response_resource.py

```diff
@@ -27,21 +25,15 @@ class SAML2ResponseResource(DirectServeHtmlResource):
     def __init__(self, hs):
         super().__init__()
         self._saml_handler = hs.get_saml_handler()
-        self._error_html_template = hs.config.saml2.saml2_error_html_template
 
     async def _async_render_GET(self, request):
         # We're not expecting any GET request on that resource if everything goes right,
         # but some IdPs sometimes end up responding with a 302 redirect on this endpoint.
         # In this case, just tell the user that something went wrong and they should
         # try to authenticate again.
-        f = failure.Failure(
-            SynapseError(400, "Unexpected GET request on /saml2/authn_response")
+        self._saml_handler._render_error(
```

Comment:
Should we make `render_error` public, or since it's still used in `synapse.rest.saml2` we consider it private?

Reply:
My thought was that since it is still used in the `saml` code it could be private so that random users of the `SamlHandler` don't use it?

Reply:
Fair yeah, I agree that the one exception here outweighs the issue of it being used accidentally elsewhere.